module/zfs/arc.c

```diff
@@ -9616,8 +9617,8 @@ l2arc_log_blk_read(l2arc_dev_t *dev,
 	}
 
 	/* Make sure the buffer checks out */
-	fletcher_4_native(this_lb,
-	    L2BLK_GET_PSIZE((this_lbp)->lbp_prop), NULL, &cksum);
+	psize = L2BLK_GET_PSIZE((this_lbp)->lbp_prop);
```

Comment:
Nit: Would it be better to move this up out of the 'Make sure the buffer checks out' section now that this is possibly used for some subsequent stuff too?

Reply:
Well we could, but the first instance we need the `psize` is in `fletcher_4_native()` that's why I put it there.